### PR TITLE
Tests for cells

### PR DESCRIPTION
**Pull request by afshin**
These are the tests for `notebook/cells` and the subsequents tweaks and bug fixes that they revealed.

|File           |% Stmts|% Branch|% Funcs|%Lines|
|---------------|------:|-------:|------:|-----:|
notebook/cells/ |  99.26|   86.62|    100| 99.62|
editor.js       |    100|   92.86|    100|   100|
index.js        |    100|      50|    100|   100|
model.js        |  98.58|   91.43|    100|   100|
widget.js       |  99.29|   82.05|    100| 99.28|

### CODE REVIEW

**Comment by afshin** [comment]
FYI, this PR is on hold because https://github.com/jupyter/jupyterlab/pull/101 will require it to be rebased and somewhat refactored.

File: test/src/notebook/cells/widget.spec.ts [hunk 1]
```diff
@@ -0,0 +1,689 @@
...
+
+    });
+
+    describe('#rendered', () => {
+
+      it('should default to true', () => {
+        let widget = new MarkdownCellWidget({ rendermime });
+        expect(widget.rendered).to.be(true);
+        requestAnimationFrame(() => {
+          expect(widget.node.classList.contains(RENDERED_CLASS)).to.be(true);
```
Comment (blink1073) [inline test/src/notebook/cells/widget.spec.ts]: Async tests need to use done()

Comment (afshin) [inline test/src/notebook/cells/widget.spec.ts]: Oops, thanks!

File: test/src/notebook/cells/editor.spec.ts [hunk 2]
```diff
@@ -3,24 +3,185 @@
...
+
+    });
+
+    describe('#edgeRequested', () => {
+
+      it('should emit a signal when the top edge is requested', () => {
+        let widget = new CellEditorWidget(new CellModel());
+        let called = false;
+        let event = generate('keydown', { keyCode: UP_ARROW });
+        widget.edgeRequested.connect(() => { called = true; });
```
Comment (blink1073) [inline test/src/notebook/cells/editor.spec.ts]: Should check the signal payload.

Comment (afshin) [inline test/src/notebook/cells/editor.spec.ts]: Good point.

**Comment by blink1073** [comment]
LGTM, thanks!

**Pull request merged**